runtime/list.go

```diff
@@ -170,6 +170,38 @@ func listCount(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
 	return seqCount(f, args[0], args[1])
 }
 
+func listRemove(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
+	if raised := checkMethodArgs(f, "remove", args, ListType, ObjectType); raised != nil {
+		return nil, raised
+	}
+	v := args[1]
+	i := -1
+	pred := func(o *Object) (bool, *BaseException) {
+		i++
+		eq, raised := Eq(f, o, v)
+		if raised != nil {
+			return false, raised
+		}
+		ret, raised := IsTrue(f, eq)
+		if raised != nil {
+			return false, raised
+		}
+		return ret, nil
+	}
+	l := toListUnsafe(args[0])
+	l.mutex.RLock()
```

Comment:
Ah, that's because seqFindFirst uses an iterator which does an RLock internally (see `listIteratorNext`). It's probably simpler anyway to do a Lock and then iterate over the elems slice directly.